### DISCUSSION TITLE
Tab Stacking Concept for Zen Browser

### DISCUSSION THREAD

**Discussion by NyllRE** [zen-browser/desktop#4077] [Ideas]
I designed a prototype for a tab stacking concept on figma.

I really think that this Tab Stacking `(also known as Tab Grouping, Tab Collection, Tab Groups)` concept would surpass other designs with the nature of knowing the tabs in the stack in a glimpse instead of something like the design that vivaldi has, it will also minimize the distraction when you are researching a specific topic that you don't really want to create a new workspace for.

This design pleases all UX requirements for a minimal browsing experience and will help people who research a ton of different concepts at the same time.

I found myself needing it because I have a workspace specifically for work I have for a client but I sometimes research a certain topic and that topic clutters the sidebar and makes it hard to differentiate between this topic and other topics I was researching earlier.

> Don't forget to Alt + Click the link :)

https://www.figma.com/proto/MVz5gtQ3SXrKXUWHYWssRy/Zen-Browser-Stack-Tabs-Proposal?node-id=1-2&t=kazSetVcrSq8qo8O-1&scaling=min-zoom&content-scaling=fixed&page-id=0%3A1&starting-point-node-id=1%3A2

**Comment by 12th-devs** [comment]
@NyllRE I would love to check out the UI/UX you have created for this, I sent a request for the figma project. I currently have a repo for the experiment Firefox tab groups that gives them a good UI/UX if you would like to check it out. I would be interested in adding some of your concepts to it once I see them, if you would allow it.

Edit: Provided Link

https://github.com/Anoms12/Advanced-Tab-Groups

**Reply by NyllRE** [reply]
I just gave you the approval, the prototype is very minimal. if required I could make it more comprehensive

**Reply by 12th-devs** [reply]
@NyllRE looks good! I am just starting to learn to code (if you consider CSS+HTML a language 😅) I am not sure if I can add the tab Icons while it is collapsed but I can make the rest of it a preference if you would like?

**Reply by NyllRE** [reply]
by the way I love your implementation. I guess the only thing missing is the tabs' favicons so you could see what exactly is being grouped

**Reply by NyllRE** [reply]
@Anoms12 for sure! I think the icons in the group tab could only be implemented natively not with a theme. perhaps someone on the Zen team could inform us

**Reply by NyllRE** [reply]
I just tried your project and I have to say, it definitely is a big game changer. like literally my whole browsing experience including all the zen browser elements have changed.

**Reply by NyllRE** [reply]
Can't help but notice that when I change workspaces they.. stay? is that a bug that you're aware of or should I open an issue?

**Reply by 12th-devs** [reply]
@NyllRE I am not having this bug and the first few lines of code should prevent that, I have had issues with it disappearing if you have any tabs open below your current tab so I turned it off, but nothing with them being persistent. Also, I have added tab ranges to the repo.

![Tab Range](https://github.com/user-attachments/assets/0b8ed116-0940-45b3-830b-fd19b15390d6)

**Comment by syscat29** [comment]
Just keep in mind that Cheff's waiting for Mozilla to finish their implementation of tab grouping first.

**Reply by cyberslot** [reply]
It would be great to have this feature ASAP. Probably a lot of Vivaldi  users (like myself and others using browsers where this function is available) would be pleased to migrate to Zen, but tab stacks is something that I personally am so accustomed (like everyone else I suppose) and that's one of the first things I've noticed is missing unfortunately.

**Reply by NyllRE** [reply]
@cyberslot i also migrated from vivaldi right to Zen which got me the idea of including this. it's one of the missing pieces that will make zen's experience a flawless organized interface
